api/agent/call.go

```diff
@@ -252,7 +253,9 @@ func (a *agent) GetCall(opts ...CallOpt) (Call, error) {
 		return nil, errors.New("no model or request provided for call")
 	}
 
-	if !a.resources.IsResourcePossible(c.Memory, uint64(c.CPUs), c.Type == models.TypeAsync) {
+	mem := c.Memory + uint64(c.TmpFsSize)
```

Comment:
I was originally on the fence about whether we should account for this, but after looking at `runc` and the kernel's `memcg` implementation, I think this is mandatory for correct behavior.  The cgroup's total page allocation count is charged against the memory limit in the memory cgroup, so if we under provision here then the user won't be able to use all of tmpfs.  The werid thing is that this lets them consume more of any pages, so if they have a piggy app they could use all of their allocation plus our extra for tmp, and then not get everything available for transient storage.  That said, file backed and regular anon pages are also charged against this total memory cap, so the behavior at the limit is going to seem strange no matter what.

Reply:
Yeah, I felt this is a bit aggressive here as well, but safer than over committing IMHO.